start

```diff
@@ -159,12 +225,42 @@ function process_args() {
       --RunTargetStackOnly)
         RUN_TARGET_STACK_ONLY="$1"
         shift
-        ;;          
+        ;;    
+      --TargetNetwork)
+        TARGET_NETWORK="$1"
+        shift
+        ;;  
+      --TargetNetworkPassphrase) 
+        TARGET_NETWORK_PASSPHRASE="$1"
+        shift
+        ;;    
+      --TargetNetworkTestAccountSecret) 
+        TARGET_NETWORK_SECRET_KEY="$1"
+        shift
+        ;;    
+      --TargetNetworkTestAccountPublic) 
+        TARGET_NETWORK_PUBLIC_KEY="$1"
+        shift
+        ;;             
       *)
       esac
     done
 }
 
+function validateHorizon () {
+    COUNTER=1
+    while ! (curl --silent --location --request GET 'http://localhost:8000' | jq --exit-status '.history_latest_ledger > 0'); do
```

Comment:
please add `set -o pipefail` at the beginning of this file, to ensure error is being propagated correctly.